packages/trader/src/Modules/SmartChart/Components/buy-toast-notification.jsx

```diff
@@ -0,0 +1,57 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { MobileWrapper, Toast, Text } from '@deriv/components';
+import { Localize } from '@deriv/translations';
+import { connect } from 'Stores/connect';
+import { findContractCategory } from '../../Trading/Helpers/contract-type';
+
+const BuyToastNotification = ({ portal_id = 'popup_root', action_toastbox, actionChangeToastbox }) => {
+    const { buy_price, currency, contract_type, list } = action_toastbox;
+    const active_trade_type = { value: contract_type };
+
+    const trade_type_name = findContractCategory(list, active_trade_type)?.contract_types?.find(
+        item => item.value === contract_type
+    ).text;
+
+    const message = (
+        <Text as='p' size='xxs' className='dc-toast__notification'>
+            <Localize
+                i18n_default_text='The purchase of <0>{{trade_type_name}} contract</0> has been completed for the amount of <0> {{buy_price}} {{currency}}</0>'
+                components={[<strong key={0} />]}
+                values={{ trade_type_name, buy_price, currency }}
+                shouldUnescape={true}
+            />
+        </Text>
+    );
+
+    React.useEffect(() => {
+        const timeout = setTimeout(() => {
+            actionChangeToastbox({});
+        }, 4000);
+
+        return () => {
+            clearTimeout(timeout);
+        };
+    }, [actionChangeToastbox, action_toastbox]);
+
+    if (!document.getElementById(portal_id) || !action_toastbox) return null;
+
+    return ReactDOM.createPortal(
+        <MobileWrapper>
+            <Toast className='dc-toast-popup-mobile' is_open={!!action_toastbox.key} timeout={0} type='notification'>
+                {message}
+            </Toast>
+        </MobileWrapper>,
+        document.getElementById(portal_id)
+    );
+};
+
+BuyToastNotification.propTypes = {
+    portal_id: PropTypes.string,
+    action_toastbox: PropTypes.bool,
+    actionChangeToastbox: PropTypes.func,
+};
+export default connect(({ modules }) => ({
+    actionChangeToastbox: modules.trade.actionChangeToastbox,
+}))(BuyToastNotification);
```

Comment:
You can leave this as a dumb component and pass this from the parent and remove `connect`

Reply:
I used the connect here because in parent the multiple connects were causing a infinite render cycle as the Notification components rendered, so i separate them in child component.
Now i have used the useStores to get the value so we don't need `connect` method anymore.
Hope this answers the question.
Thank you